### PR TITLE
Endpoints for Sprints (Meetings, Agenda Items, sprint review, sprint planning forms, and forms)

### PR DESCRIPTION
**Pull request by cherylli**
# Description

Added endpoints for sprints ( voyages/sprints on confluence) it would be more clear how this works by looking at the sprint meeting design on figma.

The implementation also takes into consideration of other forms and aim to be flexible for any future changes (questions/question types/options etc).

There are also a few small changes, like
- update `yarn format` in `package.json` to include formatting for prisma seed files
- remove package-lock.json
- swagger sorting by alphabetical order

Database 
- There are quite a lot of database updates so we "may" need a database reset if it's going to take hours to resolve. But this will be the last resort
- github prisma migrate workflow will fail due to the above
- will need a reseed for new data (a lot)
- dbdiagram has been updated. 

## Issue link
#21 but will need to keep this open as there's still voyage project, sprint check-in form submission

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested on swagger

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by kennytram — DISMISSED** [review]
lgtm

File: prisma/seed/forms/checkinform.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                                                text: "Team Channel + Team Meeting(s)"
+                                            }
+                                        ]
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    {
+                        order: 1,
```
[the remark this answers is not in the document]
Comment (cherylli) [reply prisma/seed/forms/checkinform.ts:53]: copy an paste 😂
Will update later with other changes

File: prisma/production-seed/data.ts
Comment (jenny-alexander) [inline prisma/production-seed/data.ts:1]: Small typo - I think it should be 'production-see**d**' on line 4?

File: prisma/seed/forms.ts
Comment (jenny-alexander) [inline prisma/seed/forms.ts:1]: I get lint error because getRandomDateDuringSprint isn't used. Do you think we should remove this import for now?

**Comment by jenny-alexander** [comment]
Not sure if this is a big deal, but there are some formatting issues. For example, I see a mix of single quotes and double quotes (in checkinform.ts, as an example), sometimes there aren't trailing commas. 
Do we want to make sure we all abide by the formatting in the prettier file? If yes, let's do it every time 😀. If not, I won't make comments about it anymore...but then shouldn't we scrap the prettier formatting file?

File: prisma/seed/relations.ts
Comment (jenny-alexander) [inline prisma/seed/relations.ts:1]: Should we remove imports not used?
```
import { getRandomDateDuringSprint, getSprintId } from "./utils";    
const voyages = await prisma.voyage.findMany({});
const voyageTeams = await prisma.voyageTeam.findMany({});
```

File: prisma/seed/forms.ts
Comment (cherylli) [reply prisma/seed/forms.ts:1]: yes, i think `yarn lint` doesn't scan the seed directory that's why we haven't been getting lint errors for this

**Review by kennytram — DISMISSED** [review]

**Review by curtwl — DISMISSED** [review]
All endpoints are working and successfully rejecting bad inputs with the correct error messages for me. There's one weird thing going on in Swagger for the patch route for forms though, it populates the request body input with a "response" property that has to be removed for it to work.

**Comment by cherylli** [comment]
> All endpoints are working and successfully rejecting bad inputs with the correct error messages for me. There's one weird thing going on in Swagger for the patch route for forms though, it populates the request body input with a "response" property that has to be removed for it to work.

yeah that's actually the "updateDto" response as in "response to the questions"

**Review by jenny-alexander — APPROVED** [review]
LGTM - GREAT JOB handling this complex requirement.

**Review by curtwl — APPROVED** [review]
LGTM

File: src/sprints/dto/create-agenda.dto.ts
[the remark this answers is not in the document]
Comment (cherylli) [reply src/sprints/dto/create-agenda.dto.ts:1]: answered in discord

**Pull request merged**